app/services/supplementary-billing/abstraction-billing-period.service.js

```diff
@@ -133,56 +126,86 @@ function _abstractionPeriods (billingPeriod, chargePurpose) {
     endDate: _subtractOneYear(firstPeriod.endDate)
   }
 
-  return [previousPeriod, firstPeriod]
+  const abstractionPeriods = []
+  if (_isPeriodValid(previousPeriod, billingPeriod)) {
+    abstractionPeriods.push(previousPeriod)
+  }
+
+  if (_isPeriodValid(firstPeriod, billingPeriod)) {
+    abstractionPeriods.push(firstPeriod)
+  }
+
+  return abstractionPeriods
 }
 
 /**
- * Calculates the 'billable days' for each abstraction billing period
+ * Calculates the 'billable days' for a billable abstraction period
  *
- * @param {Object[]} abstractionPeriods An array of abstraction billing periods
+ * @param {Object} abstractionPeriod An abstraction billing period
  *
- * @returns {Object[]} The array abstraction periods each with a new `billableDays` property
- * @returns {Date} periods[].startDate
- * @returns {Date} periods[].endDate
- * @returns {boolean} periods[].consider
- * @returns {number} periods[].billableDays
+ * @returns {Object} The abstraction period with a new `billableDays` property
```

Comment:
Following on from @StuAA78 original point, if we're amending the value passed in and not explicitly returning anything we shouldn't include `@returns` in our JSDoc.

```suggestion
```